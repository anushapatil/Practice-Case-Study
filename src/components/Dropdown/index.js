import React from 'react'

import Select from './styles'

const Dropdown = ({ list = [], callback, value, description, defaultId, ...restProps }) => {
  const dropdownSelected = (event) => {
    const value = event.target.value
    callback(value)
  }
  return (
    <div className='justify-content-center row mb-3'>
      <Select className='col-8 border border-secondary' onChange={dropdownSelected} {...restProps}>
        {list && list.map(item => 
          <option key={item[value]} value={item[value]} selected={defaultId === item[value]} data-testid='select-option'>
            {item[description]}
          </option>
        )}
      </Select>
    </div>
  )
}

export default Dropdown
