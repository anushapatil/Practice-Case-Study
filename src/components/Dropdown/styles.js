import styled from 'styled-components'

const Select = styled.select`
  display: inline-block;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 1.75rem 0.375rem 0.75rem;
  font-size: 1.4375rem;
  font-weight: 400;
  line-height: 1.5;
  color: #626462;
  vertical-align: middle;
  background: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3e%3cpath fill='%23343a40' d='M2 0L0 2h4zm0 5L0 3h4z'/%3e%3c/svg%3e") no-repeat right 0.75rem center/8px 10px;
  background-color: #fff;
  border: 1px solid #e3e3e0;
  border-radius: 0;
  appearance: none;
  @media (max-width: 1200px) {
    font-size: calc(1.26875rem + 0.225vw);
  }
  @media (min-width: 768px) {
    max-width: 570px;
    margin: 0 auto 1rem auto; 
  }
  @media (min-width: 992px) {
    width: 521px;
  }
  :focus {
    border-color: #51cfff;
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgba(0, 151, 208, 0.25);
  }
  :focus::-ms-value {
    color: #626462;
    background-color: #fff;
  }
  :disabled {
    color: #626462;
    background-color: #e3e3e0;
  }
`
export default Select
