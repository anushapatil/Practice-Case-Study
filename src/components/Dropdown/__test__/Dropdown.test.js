// Link.react.test.js
import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import Dropdown from '../'

const mockRoutesData = [
  {
    agency_id: 0,
    route_id: '000',
    route_label: 'Select route'
  },
  {
  agency_id: 1,
  route_id: '901',
  route_label: 'METRO Blue Line'
}]

// UI testing
test('Snapshot Dropdwons', () => {
  
  const component = renderer.create(
    <Dropdown
      list={mockRoutesData}
      value='route_id'
      description='route_label'
      callback={() => jest.fn()}
      defaultId='0'
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// DOM testing
test('Selecting option from dropdown should trigger callback', () => {
  // Setup
  const mockCallback = jest.fn()
  const { getByTestId, getAllByTestId } = render(
    <Dropdown
      list={mockRoutesData}
      value='route_id'
      description='route_label'
      callback={mockCallback}
      defaultId='0'
      data-testid='select'
    />
  )

  // Execute
  fireEvent.change(getByTestId('select'), {
    target: { value: '901' },
  })

  // Assert
  let options = getAllByTestId('select-option')
  expect(options[0].selected).toBeFalsy()
  expect(options[1].selected).toBeTruthy()
  expect(mockCallback).toHaveBeenCalledWith('901')
})

