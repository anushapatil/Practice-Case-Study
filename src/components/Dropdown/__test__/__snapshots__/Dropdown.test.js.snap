// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Dropdwons 1`] = `
<div
  className="justify-content-center row mb-3"
>
  <select
    className="sc-AykKC iTPQsP col-8 border border-secondary"
    onChange={[Function]}
  >
    <option
      data-testid="select-option"
      selected={false}
      value="000"
    >
      Select route
    </option>
    <option
      data-testid="select-option"
      selected={false}
      value="901"
    >
      METRO Blue Line
    </option>
  </select>
</div>
`;
